honeybee/radiance/datatype.py

```diff
@@ -10,61 +12,251 @@ class RadDescrDefault(object):
+                "The input for %s should be string containing the path name. " \
+                "%s was provided instead" % (self._nameString, value)
+
+            if self._expandRelative:
+                finalValue = os.path.abspath(value)
+            else:
+                finalValue = value
+
+            if self._checkExists:
+                if not os.path.exists(finalValue):
+                    raise IOError(
+                        "The specified path for %s was not found in %s" % (
+                            self._nameString, finalValue))
+
+            setattr(instance, self._name, finalValue)
+
+
+class RadInputStringFormatter:
+    """This class implements a bunch of static methods for formatting Radiance
+    input flags. I am encapsulating them within this one class for the sake of
+    clarity. The methods are static because self in this case doesn't have
+    anything to do with the operation of methods in this class.
+    """
+
+    @staticmethod
+    def fmtNormal(flagVal,inputVal):
+        """The normal format is something like -ab 5 , -ad 100, -g 0.5 0.2 0.3
+        """
+        output = " -%s"%flagVal
+
+        try:
+            output += inputVal
+        except TypeError: #raise if ip is not a str.
+            try:
+                for values in inputVal:
+                    output += " %s "%values
+            except TypeError: #raise if ip is not iterable.
+                output += " %s "%inputVal
+
+        return output
+
+
+    @staticmethod
+    def fmtJoined(flagval,inputVal):
+        """ For cases like -of , -O1 etc. where everthing after the first two
+         characters is input.
+         """
+        output = ""
+        if inputVal is not None:
+            if inputVal is True: #The value is set to be True instead of option.
+                inputVal = ''
+            output = " -%s%s "%(flagval,inputVal)
+        return output
+
+    @staticmethod
+    def fmtBool(flagVal,inputVal):
+        """If the input is just a normal toggle like -v or -h etc."""
+        output = ""
+        if inputVal is True:
+            output = "-%s "%flagVal
+        return output
+
+    @staticmethod
+    def fmtBoolDualSign(flagVal,inputVal):
+        """If the input is just a normal toggle like -v or -h etc."""
+        output = ""
+        if inputVal is True:
+            output = "+%s "%flagVal
+        elif inputVal is False:
+            output = "-%s "%flagVal
+        return output
+if __name__ == "__main__":
+    class SomeRadianceBinary(object):
+        ab = RadDescrNum('ab', 'ambinent bounces', None, [], int, True)
+        dj = RadDescrNum('dj', 'source jitter')
+        weaFile = RadDescrPath('weaFile', 'Weather File Path', True, True)
+
+        def __init__(self, ab=None, dj=None, weaFile=None):
+            self.ab = ab
+            self.dj = dj
+            self.weaFile = weaFile
+
+
+    radBinary = SomeRadianceBinary(5, 1.4, r"C:\Users\Sarith\Desktop\x.rad")
```

Comment:
@sariths I think this is the only way to assure the user is passing the right files before the execution crashes. Currently that is one of the major issues that we have in Honeybee grasshopper for input checks that's why I think it is necessary to check the extensions. Adding a new descriptors sounds great to me.